odoo/service/server.py

```diff
@@ -818,8 +822,13 @@ def run(self):
             sys.exit(1)
 
     def _runloop(self):
+        signal.pthread_sigmask(signal.SIG_BLOCK, {
+            signal.SIGXCPU,
+            signal.SIGINT, signal.SIGQUIT, signal.SIGUSR1,
+        })
         try:
             while self.alive:
+                self.set_limits()
```

Comment:
after checking with odo this can be merged back into check_limits, however check_limits needs to be moved back up so the CPU and memory limits are properly set before the first request